### PR TITLE
[feat] Visualize rows and columns group values

### PR DESCRIPTION
**Pull request by roubkar**
Added rulers like visualizations inside `BoxVirtualizer`.
Rulers' elements are also virtualized and pinned to its corresponding axis (horizontal/top, vertical/left)

### CODE REVIEW

**Review by rubenaprikyan — APPROVED** [review]
Lgtm!

**Comment by roubkar** [comment]
Resolves https://github.com/aimhubio/aim/pull/1731

File: aim/web/ui/src/modules/BaseExplorer/components/BoxVirtualizer/BoxVirtualizer.tsx [hunk 1]
```diff
@@ -85,28 +99,81 @@ function BoxVirtualizer(props: IBoxVirtualizerProps) {
...
-        ref={grid}
-        className='BoxVirtualizer__grid'
-        style={{ width: gridWidth || 0, height: gridHeight || 0 }}
+        ref={container}
+        className='BoxVirtualizer__container'
+        onScroll={onScroll}
       >
-        {Object.entries(groupedByPosition).map(props.itemsRenderer)}
+        {((columnsAxisItems && columnsAxisItems.length > 0) ||
+          (rowsAxisItems && rowsAxisItems.length > 0)) && (
```
Comment (KaroMourad) [inline aim/web/ui/src/modules/BaseExplorer/components/BoxVirtualizer/BoxVirtualizer.tsx:146]: we can simplify this expression `columnsAxisItems && columnsAxisItems.length > 0`  to `columnsAxisItems?.length`

File: aim/web/ui/src/modules/BaseExplorer/components/BoxVirtualizer/BoxVirtualizer.tsx [hunk 2]
```diff
@@ -85,28 +99,81 @@ function BoxVirtualizer(props: IBoxVirtualizerProps) {
...
+              height: gridSize.height,
+            }}
+          >
+            {rowsAxisItems?.map(props.axisItemRenderer?.rows)}
+          </div>
+        )}
+        <div
+          ref={grid}
+          style={{
+            display: 'inline',
```
Comment (KaroMourad) [inline aim/web/ui/src/modules/BaseExplorer/components/BoxVirtualizer/BoxVirtualizer.tsx:169]: let's add the class name here and include static styles in it,  remember that I have added the one named `grid` which we remove maybe after resolving the conflicts

File: aim/web/ui/src/modules/BaseExplorer/index.tsx [hunk 3]
```diff
@@ -75,7 +75,7 @@ function createExplorer(config: IExplorerConfig): () => React.ReactElement {
   };
 
   const engine = createEngine(engineConfig);
-
+  console.log(engine);
```
Comment (KaroMourad) [inline aim/web/ui/src/modules/BaseExplorer/index.tsx:78]: let's remove the `log`

File: aim/web/ui/src/modules/BaseExplorer/components/BoxVirtualizer/BoxVirtualizer.tsx (hunk 1 above)
Comment (roubkar) [reply aim/web/ui/src/modules/BaseExplorer/components/BoxVirtualizer/BoxVirtualizer.tsx:146]: `columnsAxisItems.length > 0` is more readable for me

**Review by KaroMourad — APPROVED** [review]
lgtm

**Pull request merged**